### PR TITLE
Add Graphlient::Errors::ConnectionFailedError

### PR DESCRIPTION
**Pull request by neroleung**
This is essentially https://github.com/ashkan18/graphlient/pull/59 with test.

Whenever there is an `Errno::ECONNREFUSED error`, Faraday wraps it in a `Faraday::ConnectionFailed` error. Since it inherits `Faraday::ClientError` without a response (see errors/faraday_server_error.rb), the code would return a `NoMethodError: undefined method []' for nil:NilClass` error.

So basically the fix is to rescue `Faraday::ConnectionFailed` before rescuing `Faraday::ClientError` so that we can raise a more meaningful error.

Result:
```
Graphlient::Errors::ConnectionFailedError: Failed to open TCP connection to localhost:3000 (Connection refused - connect(2) for "localhost" port 3000)
```

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,5 +1,6 @@
 ### 0.3.7 (Next)
 * Your contribution here.
+* [#68](https://github.com/ashkan18/graphlient/pull/68): Add graphlient::errors::connectionfailederror - [@neroleung](https://github.com/neroleung).
```
Comment (dblock) [inline CHANGELOG.md]: The error is `Graphlient::Errors::ConnectionFailedError`, quote it too.

File: spec/graphlient/adapters/http/faraday_adapter_spec.rb [hunk 2]
```diff
@@ -57,4 +57,25 @@
...
+    end
+
+    before do
+      wrapped_error = Errno::ECONNREFUSED.new(error_message)
+      error = Faraday::ConnectionFailed.new(wrapped_error)
+
+      stub_request(:post, url).to_raise(error)
+    end
+
+    specify do
```
Comment (dblock) [inline spec/graphlient/adapters/http/faraday_adapter_spec.rb]: Fancy :)

**Comment by dblock** [comment]
Oh one thing. You should add an UPGRADING document in this PR like [this one](https://github.com/dblock/slack-ruby-client/blob/master/UPGRADING.md) because this does change the interface. If someone was handling the Faraday error they now get a ServerError.

**Comment by neroleung** [comment]
@dblock, I've updated CHANGELOG.md and README.md. Also added UPGRADING.md. Question though, do you want a new section in README.md like https://github.com/dblock/slack-ruby-client#stable-release?

**Comment by dblock** [comment]
This is perfect, thank you. Feel free to PR something linking to UPGRADING.

**Comment by neroleung** [comment]
Thanks. This should close https://github.com/ashkan18/graphlient/issues/51, https://github.com/ashkan18/graphlient/pull/52 and https://github.com/ashkan18/graphlient/pull/59 once a new version is released.

**Comment by dblock** [comment]
I'll cut a release unless @yuki24 or @ashkan18 can beat me to it (please).

**Comment by ashkan18** [comment]
thanks @neroleung ! this is awesome! going to cut the release, hopefully before @dblock 😄

**Comment by ashkan18** [comment]
alright `0.3.7` is now published.

**Pull request merged**